apps/OpenSpace/ext/launcher/src/profile/actiondialog.cpp

```diff
@@ -0,0 +1,612 @@
+#include <QLabel>
+#include <QLineEdit>
+#include <QListWidget>
+#include <QMessageBox>
+#include <QPushButton>
+#include <QTextEdit>
+#include <QVBoxLayout>
+
+using namespace openspace;
+
+namespace {
+    void updateListItem(QListWidgetItem* item, const Profile::Action& action) {
+        ghoul_assert(item, "Item must exist at this point");
+        item->setText(
+            action.name.empty() ?
+            QString::fromStdString(action.identifier) :
+            QString::fromStdString(action.name)
+        );
+    }
+
+    void updateListItem(QListWidgetItem* item, const Profile::Keybinding& kb) {
+        ghoul_assert(item, "Item must exist at this point");
+        std::string name = fmt::format("{}\t{}", ghoul::to_string(kb.key), kb.action);
+        item->setText(QString::fromStdString(name));
+    }
+} // namespace
+
+ActionDialog::ActionDialog(Profile& profile, QWidget* parent)
+    : QDialog(parent)
+    , _profile(profile)
+{
+    _actions.data = _profile.actions;
+    _keybindings.data = _profile.keybindings;
+
+    setWindowTitle("Actions and Keybindings");
+    createWidgets();
+}
+
+void ActionDialog::createWidgets() {
+    //         Column 0            Column 1         Column 2
+    //  *----------------------*---------------*----------------*
+    //  |                      | Identifier    | [oooooooooooo] |    Row 0
+    //  |                      | Name          | [oooooooooooo] |    Row 1
+    //  |                      | GUI Path      | [oooooooooooo] |    Row 2
+    //  |                      | Documentation | [oooooooooooo] |    Row 3
+    //  |                      | Is Local      | []             |    Row 4
+    //  |                      | Script        | [oooooooooooo] |    Row 5
+    //  *----------------------*---------------*----------------*
+    //  | [+] [-]              |               | [Save] [Cancel]|    Row 6
+    //  *----------------------*---------------*----------------*
+    //  |=======================================================|    Row 7
+    //  *----------------------*---------------*----------------|
+    //  |                      | Modifier      | []S []C []A    |    Row 8
+    //  |                      | Key           | DDDDDDDDDDDD>  |    Row 9
+    //  |                      | Actions       | DDDDDDDDDDDD>  |    Row 10
+    //  |                      | Add action    | [oooooooooooo] |    Row 11
+    //  |                      |               |                |    Row 12
+    //  *----------------------*---------------*----------------*
+    //  | [+] [-]              |               | [Save] [Cancel]|    Row 13
+    //  *----------------------*---------------*----------------*
+    //  |=======================================================|    Row 14
+    //  *----------------------*---------------*----------------*
+    //  |                                      | [Save] [Cancel]|    Row 15
+    //  *----------------------*---------------*----------------*
+
```

Comment:
This is a good idea for showing what the intended layout is

Reply:
Done